src/Tapper/TypeMappers/DictionaryTypeMappers.tt

```diff
@@ -29,14 +29,14 @@ public class <#= type.Name.Replace("`", null) #>TypeMapper : ITypeMapper
 
     public <#= type.Name.Replace("`", null) #>TypeMapper(Compilation compilation)
     {
-        Assign = compilation.GetTypeByMetadataName("<#= type.FullName #>")!;
+        Assign = compilation.GetTypeByMetadataName("<#= type.FullName #>")!.GetUnboundedType();
     }
 
     public string MapTo(ITypeSymbol typeSymbol, ITranspilationOptions options)
     {
         if (typeSymbol is INamedTypeSymbol namedTypeSymbol
             && namedTypeSymbol.IsGenericType
-            && SymbolEqualityComparer.Default.Equals(namedTypeSymbol.ConstructedFrom, Assign))
+            && SymbolEqualityComparer.Default.Equals(namedTypeSymbol.GetUnboundedType(), Assign))
```

Comment:
```suggestion
            && SymbolEqualityComparer.Default.Equals(namedTypeSymbol, Assign))
```

Reply:
your suggestion was missing the `.ConstructedFrom` that has been there before my changes.